tests/expressions/tests.py

```diff
@@ -387,6 +387,125 @@ def test_ticket_18375_chained_filters(self):
+                'is_ceo_small_2': False,
+                'largest_company': None
+            },
+            {
+                'firstname': 'Frank',
+                'is_point_of_contact': False,
+                'is_not_point_of_contact': True,
+                'is_ceo_of_small_company': True,
+                'is_ceo_small_2': True,
+                'largest_company': 'Foobar Ltd.',
+            },
+            {
+                'firstname': 'Joe',
+                'is_point_of_contact': True,
+                'is_not_point_of_contact': False,
+                'is_ceo_of_small_company': False,
+                'is_ceo_small_2': False,
+                'largest_company': 'Example Inc.',
+            },
+            {
+                'firstname': 'Max',
+                'is_point_of_contact': False,
+                'is_not_point_of_contact': True,
+                'is_ceo_of_small_company': True,
+                'is_ceo_small_2': True,
+                'largest_company': 'Example Inc.'
+            }
+        ], list(qs))
+
+        self.assertEqual(1, qs.values('pk').filter(is_point_of_contact=True).count())
+        # Sanity check: another (less elegant) way to write the same query: this will use a
+        # LEFT OUTER JOIN and an IS NULL, which is probably less efficient than an EXISTS()
+        self.assertEqual(1, Employee.objects.exclude(company_point_of_contact_set=None).values('pk').count())
+
+    def test_in_subquery(self):
+        # This is a contrived test (and you really wouldn't write this query), but it
+        # is a succint way to test the __in=SubQuery() construct.
+        small_companies = Company.objects.filter(num_employees__lt=200).values('pk')
+        subquery_test = Company.objects.filter(pk__in=SubQuery(small_companies))
+
+        self.assertEqual(2, subquery_test.count())
+
+        subquery_test2 = Company.objects.filter(pk=SubQuery(small_companies.filter(num_employees=3)))
+        self.assertEqual(1, subquery_test2.count())
+
+    def test_nested_subquery(self):
+        inner = Company.objects.filter(point_of_contact=OuterRef('pk'))
+        outer = Employee.objects.annotate(is_point_of_contact=Exists(inner))
+        contrived = Employee.objects.annotate(
+            is_point_of_contact=SubQuery(outer.filter(pk=OuterRef('pk')).values('is_point_of_contact'),
+                                         output_field=models.BooleanField()))
+
+        self.assertEqual(sorted(outer.values_list()), sorted(contrived.values_list()))
+
+    def test_nested_subquery_outer_ref_2(self):
+        first = Time.objects.create(time='09:00')
+        second = Time.objects.create(time='17:00')
+        third = Time.objects.create(time='21:00')
+        SimulationRun.objects.bulk_create([
+            SimulationRun(start=first, end=second, midpoint='12:00'),
+            SimulationRun(start=first, end=third, midpoint='15:00'),
+            SimulationRun(start=second, end=first, midpoint='00:00')
```

Comment:
Include a trailing comma so that if more items are added later, so we don't need to modify this line again.